pkg/controllers/dynakube/kspm/reconciler.go

```diff
@@ -0,0 +1,41 @@
+package kspm
+
+import (
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ReconcilerBuilder func(
+	client client.Client,
+	apiReader client.Reader,
+	dk *dynakube.DynaKube,
+) controllers.Reconciler
+
+func NewReconciler( //nolint
+	client client.Client,
+	apiReader client.Reader,
+	dk *dynakube.DynaKube,
+) controllers.Reconciler {
+	return &Reconciler{
+		client:    client,
+		apiReader: apiReader,
+		dk:        dk,
+	}
+}
+
+type Reconciler struct {
+	client    client.Client
+	apiReader client.Reader
+	dk        *dynakube.DynaKube
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context) error {
+	if r.dk.IsKSPMEnabled() {
+		return ensureKSPMSecret(ctx, r.client, r.apiReader, r.dk)
+	} else {
+		return removeKSPMSecret(ctx, r.client, r.apiReader, r.dk)
+	}
+}
```

Comment:
There should be a `kspm` reconciler, but there will be other sub-reconcilers here as well, not just the token

the structure should be that the `kspm` reconciler just calls the other sub-reconcilers.
you could have a `pkg/controllers/dynakube/kspm/token` package, where this code is, and you just call that `token` reconciler in the `kspm` reconciler

Reply:
example: https://github.com/Dynatrace/dynatrace-operator/blob/acade6bbb33c5a0f6444c1230d5091d7521e0700/pkg/controllers/dynakube/logmodule/reconciler.go#L28-L62

Reply:
done